### PR TITLE
169 improve apidoc generation

### LINKED ISSUE #169: Improve apidoc generation
- Generated module titles can be very long. `revolve2.core.optimization.ea.openai_es module` could be `openai_es module`.
- More like this?

### PR DESCRIPTION
**Pull request by surgura**
* Remove autodoc and apidoc in favor of autoapi.
* No need to manually edit api documentation ever again.
* Display not perfect but reasonable for now. See #341 

closes #169

### CODE REVIEW

File: docs/source/index.rst
Comment (surgura) [inline docs/source/index.rst:1]: fyi api reference is programmatically added by autoapi

File: experimentation/revolve2/experimentation/_util/__init__.py
Comment (surgura) [inline experimentation/revolve2/experimentation/_util/__init__.py:1]: this file was missing but we didn't see it until now

File: experimentation/revolve2/experimentation/optimization/ea/selection/_multiple_unique.py
Comment (surgura) [inline experimentation/revolve2/experimentation/optimization/ea/selection/_multiple_unique.py:1]: changed the generic parameters so they don't class with the `Individual` class in the docs

File: docs/source/_templates/autoapi/python/module.rst
Comment (surgura) [inline docs/source/_templates/autoapi/python/module.rst:1]: very similar to the default used by autoapi, just added that module names don't include their complete path package1.package2.module adn just `module`. it's implicit anyway wrt to where you are in the docs

**Review by oliverweissl — APPROVED** [review]
I assume the failed CI is wanted because the system changed

**Comment by surgura** [comment]
> I assume the failed CI is wanted because the system changed

not intentional, i'll fix it before merging

**Pull request merged**